src/rest_framework_dso/embedding.py

```diff
@@ -7,89 +7,212 @@
 from __future__ import annotations
 
 from collections import defaultdict
+from dataclasses import dataclass
+from functools import cached_property
 from itertools import islice
-from typing import Callable, Dict, Iterable, Iterator, List, Optional, TypeVar, Union
+from typing import Callable, Dict, Iterable, Iterator, List, Optional, Type, TypeVar, Union
 
 from django.db import models
 from lru import LRU
 from rest_framework import serializers
-from rest_framework.exceptions import ParseError, PermissionDenied
+from rest_framework.exceptions import ParseError
 
-from dso_api.dynamic_api.permissions import fetch_scopes_for_model
-from rest_framework_dso.fields import AbstractEmbeddedField, EmbeddedField
+from rest_framework_dso.fields import AbstractEmbeddedField
 from rest_framework_dso.serializer_helpers import ReturnGenerator, peek_iterable
 
-EmbeddedFieldDict = Dict[str, AbstractEmbeddedField]
+DictOfDicts = Dict[str, Dict[str, dict]]
```

Comment:
`dict` -> `Dict` (in the embedded dict).

Reply:
It can't use recursion, so I think it's better to leave it like this. `Dict[str, Dict[str, Dict[str, ...]]]` would not be readable.